src/Lecture2.hs

```diff
@@ -198,7 +319,13 @@ verify that.
 [1,2,3,4,7]
 -}
 merge :: [Int] -> [Int] -> [Int]
-merge = error "TODO"
+merge [] [] = []
+merge [] l = l
+merge l [] = l
+merge (x1: xs1) (x2: xs2)
+  | x1 > x2 = x2 : merge (x1: xs1) xs2
+  | x1 < x2 = x1 : merge xs1 (x2: xs2)
+  | otherwise = [x1, x1] ++ merge xs1 xs2
```

Comment:
Nice optimization for the case of equal heads! 👏🏻 
You can use the `:` operator instead of `++` to avoid creating temporary lists:

```suggestion
  | otherwise = x1 : x1 : merge xs1 xs2
```